### DISCUSSION TITLE
Advanced Scripting Missing

### DISCUSSION THREAD

**Discussion by inspirz** [TabularEditor/TabularEditor3#1263] [🙋♀️TE3 Q&A]
Hi,

I downloaded the 30-day trial version of the Tabular Editor so I could use Advanced Scripting to analyze a Power BI report. However, I don't see the Advanced Scripting tab, does the trial version have it? Or is there some options I need to select to see it?

Thanks!

**Comment by otykier** [comment]
Check [this](https://docs.tabulareditor.com/onboarding/cs-scripts-and-macros.html#creating_a_script) out :-)

**Reply by inspirz** [reply]
Hi,

Thanks a lot. In the website I pasted below there's a section called "Loading the Rules" and the author provides a snippet of code that analyzes your Power BI performance. I was hoping for Advanced Scripting to use that. Would you know how I can adapt that code below to run in C#?

https://powerbi.microsoft.com/en-us/blog/best-practice-rules-to-improve-your-models-performance/


System.Net.WebClient w = new System.Net.WebClient(); 

string path = System.Environment.GetFolderPath(System.Environment.SpecialFolder.LocalApplicationData);
string url = "https://raw.githubusercontent.com/microsoft/Analysis-Services/master/BestPracticeRules/BPARules.json";
string downloadLoc = path+@"\TabularEditor\BPARules.json";
w.DownloadFile(url, downloadLoc);

**Reply by inspirz** [reply]
Oh I think I figured it out, I can use the string URL in the Best Practice Analyzer...

**Reply by otykier** [reply]
Yes, that should work. But also, Advanced Scripting is the same as C# Scripting :-)
